napari/utils/settings/_defaults.py

```diff
@@ -196,10 +196,13 @@ class PluginsSettings(BaseNapariSettings):
     """Plugins Settings."""
 
     schema_version = (0, 1, 0)
-    plugins_call_order: List[str] = Field(
-        [],
-        title=trans._("Plugin call order"),
-        description=trans._("Sort plugins call order"),
+
+    plugins_call_order: list = Field(
```

Comment:
whenever possible, I think it'd be great to keep our settings schema as strictly-typed as possible.  So rather than `plugins_call_order: list`, this should be `plugins_call_order: List[Tuple[str, str, bool]]`  (otherwise someone has to go figure out where and how this gets set... to figure out what it's supposed to be).

However, I'd like to propose that we arrange this a bit differently.  Currently, if you look at `settings.yaml`, it reads like this:

```yaml
plugins:
  plugins_call_order:
  - - napari_experimental_provide_dock_widget
    - skimage.filters
    - true
  - - napari_get_reader
    - ome_zarr
    - true
  - - napari_get_reader
    - builtins
    - true
  - - napari_get_writer
    - builtins
    - true
```

All of the hooks are mixed in together... I think it would be clearer if each hook was it's own sub-dict:

```yaml
plugins:
  call_order:
    napari_experimental_provide_dock_widget:
      - plugin: skimage.filters
        enabled: true
    napari_get_reader:
      - plugin: ome_zarr
        enabled: true
      - plugin: builtins
        enabled: true
    napari_get_writer:
      - plugin: builtins
        enabled: true
```

Additionally, I think both the new `QtPluginSorter.values` and `plugins.load_plugin_manager_settings` functions belong as methods on the `plugin_manager` itself.  So I'd proposed making a subclass in `plugins.__init__` (we can push it upstream to napari-plugin-engine later).  Something like this:

```python
from typing import TypedDict, List, Dict
from napari_plugin_engine import PluginManager as _PM

class PluginHookOption(TypedDict):
    plugin: str
    enabled: bool


CallOrderDict = Dict[str, List[PluginHookOption]]


class PluginManager(_PM):
    # replaces QtPluginSorter.values
    def call_order(self) -> CallOrderDict:
        return {
            spec_name: [
                {'plugin': impl.plugin_name, 'enabled': impl.enabled}
                for impl in reversed(caller.get_hookimpls())
            ]
            for spec_name, caller in self.hooks.items()
        }

    # replaces plugins.load_plugin_manager_settings
    def set_call_order(self, new_order: CallOrderDict):
        for spec_name, hook_caller in self.hooks.items():
            spec_order = new_order.get(spec_name)
            order = []
            for p in spec_order:
                order.append(p['plugin'])
                hook_caller._set_plugin_enabled(p['plugin'], p['enabled'])
```

after that, the type annotation on `PluginsSettings.call_order` can be `plugins.CallOrderDict`

I haven't had a chance to look much deeper into the PR yet, but those are my first thoughts...

Reply:
thanks for the feedback, @tlambert03 !  I will work on moving these things around.

Reply:
I made these changes.  Thanks for the input!